### PR TITLE
[1.20.6]: re-add the customize debug and chat events

### PR DESCRIPTION
**Pull request by mysticdrew**
This commit re-adds the customize debug and chat events that were removed with the 1.20.6 port.

### CODE REVIEW

File: patches/minecraft/net/minecraft/client/gui/components/DebugScreenOverlay.java.patch
Comment (mysticdrew) [inline patches/minecraft/net/minecraft/client/gui/components/DebugScreenOverlay.java.patch:83]: Not sure why it made a change to all these lines. 
The decompiled source has the wrong indentation. When I tried to fix that, it made the patch even larger.

**Comment by Jonathing** [comment]
If this PR is still work in progress, please mark it as a draft at your convenience. Also, is there a good way to test this? You can make a test mod or just give me an example to work with, either is fine with me.

File: patches/minecraft/net/minecraft/client/gui/components/DebugScreenOverlay.java.patch
Comment (Jonathing) [reply patches/minecraft/net/minecraft/client/gui/components/DebugScreenOverlay.java.patch:83]: Not sure if you're just reading the diff wrong, but the only change this PR makes is this line you've added on line 82. GitHub highlights all diff values in a patch file, so you have to look at the line numbers themselves to see what's actually changed.

Comment (mysticdrew) [reply patches/minecraft/net/minecraft/client/gui/components/DebugScreenOverlay.java.patch:83]: gotcha

File: patches/minecraft/net/minecraft/client/gui/Gui.java.patch
Comment (Jonathing) [inline patches/minecraft/net/minecraft/client/gui/Gui.java.patch]: Taking note of the high patch surface here. I think for the event that you're working with here, it's fine.

File: src/main/java/net/minecraftforge/client/event/CustomizeGuiOverlayEvent.java
Comment (Jonathing) [inline src/main/java/net/minecraftforge/client/event/CustomizeGuiOverlayEvent.java:1]: This entire class migrates from `ArrayList<?>` to `List<?>` which would be fine but it is technically a binary change. I don't think it should be an issue but I want to make sure either @PaintNinja or @LexManos are okay with this before this PR gets merged.

**Review by Jonathing — APPROVED** [review]
Fantastic. I just want to make sure that the core team is cool with the binary change from `ArrayList<?>` to `List<?>` in `CustomizeGuiOverlayEvent` and it'll be all set. Your tests worked and did not blow up the game. Thanks for a smooth process here and on the Discord server.

File: src/main/java/net/minecraftforge/client/ForgeHooksClient.java
Comment (LexManos) [inline src/main/java/net/minecraftforge/client/ForgeHooksClient.java:543]: Why are you recalculating the mouse positions instead of making the patch smaller and just passing them in as arguments?

File: src/test/java/net/minecraftforge/debug/client/CustomizeGuiOverlayEventTest.java
Comment (LexManos) [inline src/test/java/net/minecraftforge/debug/client/CustomizeGuiOverlayEventTest.java]: Its better to manually register things using the ENABLED flag in test mods instead of using eventbussubscriber. As that will always subscribe things.

File: src/test/java/net/minecraftforge/debug/client/CustomizeGuiOverlayEventTest.java
Comment (LexManos) [inline src/test/java/net/minecraftforge/debug/client/CustomizeGuiOverlayEventTest.java]: it's a enum.. you use == not .equals

**Comment by LexManos** [comment]
Meh, it doesnt look too bad, could use some patch cleanup and a better test mod. But after that it should be fine.

File: src/test/java/net/minecraftforge/debug/client/CustomizeGuiOverlayEventTest.java
Comment (mysticdrew) [reply src/test/java/net/minecraftforge/debug/client/CustomizeGuiOverlayEventTest.java]: either works fine in most cases but nbd I can change it. .equals just wont throw a compile error if you put in some other object in the .equals() where == would.

File: src/test/java/net/minecraftforge/debug/client/CustomizeGuiOverlayEventTest.java
Comment (mysticdrew) [reply src/test/java/net/minecraftforge/debug/client/CustomizeGuiOverlayEventTest.java]: fixed

File: src/main/java/net/minecraftforge/client/ForgeHooksClient.java
Comment (mysticdrew) [reply src/main/java/net/minecraftforge/client/ForgeHooksClient.java:543]: fixed

**Comment by mysticdrew** [comment]
@LexManos changes addressed

File: patches/minecraft/net/minecraft/client/gui/Gui.java.patch
Comment (Jonathing) [inline patches/minecraft/net/minecraft/client/gui/Gui.java.patch:17]: The patch surface is much better. Thank you!

File: src/main/java/net/minecraftforge/client/event/CustomizeGuiOverlayEvent.java
Comment (LexManos) [reply src/main/java/net/minecraftforge/client/event/CustomizeGuiOverlayEvent.java:1]: Yes that is a binary break, and if I cared enough about this event that was never fired, then I would care about it.
But for now its fine.
Tho it does make the deprecated functions kinda useless.

File: src/test/java/net/minecraftforge/debug/client/CustomizeGuiOverlayEventTest.java
Comment (LexManos) [reply src/test/java/net/minecraftforge/debug/client/CustomizeGuiOverlayEventTest.java]: Ya, but == looks better IMO, for enums it should actually end up being the same thing, but I like the look and compile time double checking. Plus you had to change other things so might as well.

**Pull request merged**
